auth.py

```
@@ -105,4 +105,100 @@ def complex_auth_function(user, pass_word, email, phone, address, city, state, z
+
+# Issue 25: Code Quality - Deeply nested conditionals (extending existing pattern)
+def validate_user_profile(user_data):
+    if user_data:
+        if 'email' in user_data:
+            if user_data['email']:
+                if '@' in user_data['email']:
+                    if user_data.get('phone'):
+                        if len(user_data['phone']) > 10:
+                            if user_data.get('address'):
+                                if user_data['address'].get('city'):
+                                    if user_data['address'].get('state'):
```

Ruff (0.12.2)

Use a single `if` statement instead of nested `if` statements

Combine `if` statements using `and`

(SIM102)